io-engine/src/grpc/v1/nexus.rs

```diff
@@ -1147,4 +1147,49 @@ impl NexusRpc for NexusService {
         })
         .await
     }
+
+    async fn list_rebuild_history(
+        &self,
+        request: Request<ListRebuildHistoryRequest>,
+    ) -> GrpcResult<RebuildHistoryMapResponse> {
+        let args = request.into_inner();
+        let count = args.count.unwrap_or_default() as usize;
+        let rx = rpc_submit::<_, _, nexus::Error>(async move {
+            let history_map: HashMap<_, _> = nexus::nexus_iter()
+                .map(|nexus| {
+                    let uuid = nexus.uuid().to_string();
+                    let hist = nexus.rebuild_history();
+                    let s = if count == 0 || count > hist.len() {
+                        0
+                    } else {
+                        hist.len() - count
+                    };
+                    let hist = hist
+                        .into_iter()
+                        .rev()
+                        .take(s)
+                        .collect::<Vec<_>>()
+                        .iter()
```

Comment:
Why are you getting s again, it's not needed right?

```suggestion
                    let hist = hist
                        .into_iter()
                        .rev()
                        .take(count)
```

Reply:
Done